src/core/pdf/base/XojPdfExport.cpp

```diff
@@ -1,12 +1,227 @@
+                              exportBackground <= EXPORT_BACKGROUND_UNRULED);
+    } else {
+        view.drawPage(p, this->cr, true /* dont render eraseable */, true /* don't rerender the pdf background */,
+                      exportBackground == EXPORT_BACKGROUND_NONE, exportBackground <= EXPORT_BACKGROUND_UNRULED);
+    }
+
+    // next page
+    cairo_show_page(this->cr);
+    cairo_restore(this->cr);
+}
+
+// export layers one by one to produce as many PDF pages as there are layers.
+void XojPdfExport::exportPageLayers(size_t page) {
+    PageRef p = doc->getPage(page);
+
+    // We keep a copy of the layers initial Visible state
+    std::map<Layer*, bool> initialVisibility;
+    for (const auto& layer: *p->getLayers()) {
+        initialVisibility[layer] = layer->isVisible();
+        layer->setVisible(false);
+    }
+
+    // We draw as many pages as there are layers. The first page has
+    // only Layer 1 visible, the last has all layers visible.
+    for (const auto& layer: *p->getLayers()) {
+        layer->setVisible(true);
+        exportPage(page);
+    }
+
+    // We restore the initial visibilities
+    for (const auto& layer: *p->getLayers()) layer->setVisible(initialVisibility[layer]);
+}
+
+auto XojPdfExport::createPdf(const PageRangeVector& range, bool progressiveMode) -> bool {
+    if (range.empty()) {
+        this->lastError = _("No pages to export!");
+        return false;
+    }
+
+    if (!createCairoCr()) {
+        this->lastError = _("Failed to initialize PDF Cairo surface");
+        this->lastError += "\nCairo error: ";
+        this->lastError += cairo_status_to_string(cairo_surface_status(this->surface));
+        return false;
+    }
+
+    size_t count{0};
+    for (const auto& e: range) { count += e.last - e.first + 1; }
+
+    if (progressListener)
+        progressListener->setMaximumState(static_cast<int>(count));
+
+    count = 0;
+    for (const auto& e: range) {
+        auto max = std::min(e.last, doc->getPageCount());
+        for (size_t i = e.first; i <= max; i++) {
+            if (progressiveMode) {
+                exportPageLayers(i);
+            } else {
+                exportPage(i);
+            }
+
+            if (progressListener) {
+                progressListener->setCurrentState(static_cast<int>(count));
+				++count;
+            }
+        }
+    }
+
+    freeCairoResources();
+    return true;
+}
+
+auto XojPdfExport::createPdf(bool progressiveMode) -> bool {
+    if (doc->getPageCount() < 1) {
+        lastError = _("No pages to export!");
+        return false;
+    }
+
+    if (!createCairoCr()) {
+        this->lastError = _("Failed to initialize PDF Cairo surface");
+        this->lastError += "\nCairo error: ";
+        this->lastError += cairo_status_to_string(cairo_surface_status(this->surface));
+        return false;
+    }
+
+    size_t count = doc->getPageCount();
+    if (progressListener)
+        progressListener->setMaximumState(static_cast<int>(count));
```

Comment:
```suggestion
    if (progressListener) {
        progressListener->setMaximumState(static_cast<int>(count));
    }
```
See https://github.com/xournalpp/xournalpp/wiki/Coding-conventions#indentation

Reply:
All respective expressions should have braces now.